armi/reactor/blueprints/assemblyBlueprint.py

```diff
@@ -153,17 +192,67 @@ def _constructAssembly(self, cs, blueprint):
 
         return a
 
-    def _createBlock(self, cs, blueprint, bDesign, axialIndex):
+    def _compileAssemblyMaterialInput(self):
+        """
+        Combine the blockwise and componentwise material inputs for the entire
+        assembly into a single dict `assemblyMaterialInput`. The combined dict
+        has the structure:
+        ```
+        {"": {blockwise_mod_name1: blockwise_mod_list1}, ..., {blockwise_mod_nameN: blockwise_mod_listN},
+         comp_name1: {componentwise_mod_name1: componentwise_mod_list1}, ..., {componentwise_mod_nameN: componentwise_mod_listN},
+         .
+         .
+         .
+         comp_nameN: {componentwise_mod_name1: componentwise_mod_list1}, ..., {componentwise_mod_nameN: componentwise_mod_listN}}
+         ```
+
+        For each `mod_list`, there are expected to be as many entries as there are
+        blocks in the assembly. This may be relaxed in the future by allowing a
+        single value to apply to all blocks without needing to write out a whole list.
+
+        To maintain backwards compatibility, we first assumed that there might be
+        blockwise matMods on the base object, then we add in any blockwise matMods
+        that are explicitly denoted using the `blockwise` attribute.
+        """
+        assemblyMaterialInput = {}
+
+        if self.materialModifications != None:
+            assemblyMaterialInput[""] = {}
+            for modName, modList in self.materialModifications.items():
+                assemblyMaterialInput[""][modName] = modList
+
+            if self.materialModifications.blockwise:
+                for modName, modList in self.materialModifications.blockwise.items():
+                    if modName in assemblyMaterialInput[""].keys():
+                        raise KeyError(
+                            f"{modName} material modification is being specified blockwise"
+                            f" twice in the {self.name} assembly."
+                        )
+                    assemblyMaterialInput[""][modName] = modList
+
+            if self.materialModifications.componentwise:
+                for component, mods in self.materialModifications.componentwise.items():
+                    assemblyMaterialInput[component] = {}
+                    for modName, modList in mods.items():
+                        assemblyMaterialInput[component][modName] = modList
+
+        return assemblyMaterialInput
```

Comment:
Why not just modify the object's `self.materialModifications` data structure directly here rather than making a modified copy of the data and passing it around in function calls?

Reply:
I changed the flow to get rid of the `_compileAssemblyMaterialInput` method entirely. So no duplicates are made and nothing is passed around anymore.

Reply:
Since the changes were made (possibly to address the comments directly), I can't really see or understand this conversation any more.  Keckler's comment makes it sound like a good change was made here.  One of you two should either continue this conversation or "resolve" it.

Reply:
Resolving now.